ttnn/cpp/ttnn/tensor/layout/tensor_layout.hpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+
+#include "alignment.hpp"
+#include "size.hpp"
+#include "page_config.hpp"
+
+#include "ttnn/tensor/enum_types.hpp"
+#include "ttnn/tensor/types.hpp"
+#include "ttnn/tensor/shape/shape.hpp"
+
+namespace tt::tt_metal {
+
+using Strides = std::vector<size_t>;
```

Comment:
I think Strides should also be ShapeBase. This way it would both support negative indexes and use SmallVector

Reply:
Let me evolve this as we go, strides is a stretch right now, there are only anticipated users outside of tests